src/routes/auth/index.ts

```diff
@@ -1,140 +1,286 @@
+			return context.json({ message: 'Invalid profile id or Account not activated' } satisfies StatusResponse, StatusCodes.UNAUTHORIZED);
 		}
-		throw error;
-	}
 
-	const { id: profileId, password: storedPassword } = await getProfileIdPassword(context.env.database, parsedBody.email);
-	if (!profileId || !storedPassword) {
-		return context.json<StatusResponse>({ message: 'Invalid email' }, StatusCodes.UNAUTHORIZED);
-	}
+		const accessLevel = await getAccessLevel(context.env.database, profileId);
+		if (!accessLevel) {
+			return context.json({ message: 'Access not found' } satisfies StatusResponse, StatusCodes.NOT_FOUND);
+		}
 
-	const isProfileValid = await checkProfile(context.env.database, profileId);
-	if (!isProfileValid) {
-		return context.json<StatusResponse>({ message: 'Invalid profile id or Account not activated' }, StatusCodes.UNAUTHORIZED);
-	}
+		const isValid = await validatePassword(parsedBody.password, storedPassword);
+		if (!isValid) {
+			return context.json({ message: 'Invalid email or password' } satisfies StatusResponse, StatusCodes.UNAUTHORIZED);
+		}
 
-	const accessLevel = await getAccessLevel(context.env.database, profileId);
-	if (!accessLevel) {
-		return context.json<StatusResponse>({ message: 'Access not found' }, StatusCodes.NOT_FOUND);
-	}
+		const sessionToken = crypto.randomUUID();
+		const hoursOffset = 24;
+		const tokenExpiryISO = addHours(new Date(), hoursOffset).toISOString();
+		const sessionDataObject: SessionData = {
+			id: profileId,
+			email: parsedBody.email,
+			access: accessLevel,
+			expiry: tokenExpiryISO
+		};
+		const sessionData = JSON.stringify(sessionDataObject);
+		await context.env.SESSION_TOKENS.put(sessionToken, sessionData);
 
-	const isValid = await validatePassword(parsedBody.password, storedPassword);
-	if (!isValid) {
-		return context.json<StatusResponse>({ message: 'Invalid email or password' }, StatusCodes.UNAUTHORIZED);
+		context.header('Set-Cookie', `auth_token=${sessionToken}; HttpOnly; Secure; SameSite=Strict; Expires=${tokenExpiryISO}; Path=/;`);
+
+		return context.json({ message: 'Sign in successful' } satisfies StatusResponse, StatusCodes.CREATED);
 	}
+);
 
-	const sessionToken = crypto.randomUUID();
-	const hoursOffset = 24;
-	const tokenExpiryISO = addHours(new Date(), hoursOffset).toISOString();
-	const sessionDataObject: SessionData = {
-		id: profileId,
-		email: parsedBody.email,
-		access: accessLevel,
-		expiry: tokenExpiryISO
-	};
-	const sessionData = JSON.stringify(sessionDataObject);
-	await context.env.SESSION_TOKENS.put(sessionToken, sessionData);
-
-	context.header('Set-Cookie', `auth_token=${sessionToken}; HttpOnly; Secure; SameSite=Strict; Expires=${tokenExpiryISO}; Path=/;`);
-
-	return context.json(sessionToken);
+// Protected auth routes
+export const protectedAuthRoutes = new OpenAPIHono<EnvironmentBindings>({
+	defaultHook: statusResponseFormatter
 });
+
+protectedAuthRoutes.openapi(
+	createRoute({
+		method: 'get',
+		path: '/',
```

Comment:
```suggestion
		path: '/heartbeat',
```

The way it is right now, that route would be: `GET /auth/`